### ISSUE TITLE
Get 123done.org set up for subscription services

### ISSUE THREAD

**Issue by clouserw** [mozilla/fxa#1358]
We should update 123done.org to work with subscription services!

**Comment by lmorchard** [comment]
We might want to piggy-back on this and come up with a checklist of everything that needs doing for a subscription service RP in general. Off the top of my head, many of these are owned by multiple teams:

* Create a product & plan in Stripe, get IDs for both (IT?)
* Create an oauth client for the RP, get the ID (ops?)
* For fxa-auth-server (ops?)
  * Modify [`SUBSCRIPTTION_PRODUCT_CAPABILITIES`](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/config/index.js#L636) config ([example](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/config/dev.json#L93)) to associate new capability names with the product ID.
  * Modify [`SUBSCRIPTIONS_CLIENT_CAPABILITIES`](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/config/index.js#L642) config ([example](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/config/dev.json#L100)) associate the new oauth client ID with the new product capability names.
* Ensure the subscription RP (e.g. 123Done) looks for the new product capability in the user's profile to enable subscription-gated features.

The confusing thing here might be product ID vs product capabilities. I've been meaning to write up an ADR explaining the thoughts behind this. 

TL;DR: One product can enable multiple services. Different products could offer different combinations of services as we see fit for business or marketing reasons.

However, we don't want every RP to have to know about product bundles. So, the "capabilities" provided by each individual RP should be uniquely named. Then, we can associate named capabilities with one or more products. And we can associate RPs with named capabilties. Then, when a user subscribes to a product, we can tell individual RPs which specific capabilities are enabled without exposing the product.

**Comment by lmorchard** [comment]
FWIW: I submitted a PR with that ADR mentioned earlier. https://github.com/mozilla/fxa/pull/1385

**Comment by clouserw** [comment]
Thanks for writing that ADR.

A checklist is a great idea and we'll need it anyway.  We have a meeting on Friday with the secure proxy folks to talk about requirements in both directions.

**Comment by lmorchard** [comment]
I think this is still kind of blocked by #1723 and #1364.

We closed #926, which will at least allow us to stick a product subscription link into 123done that will work for an already-signed-in user. But, that's not the "enter email to signin/signup" form (#1364) and won't carry through a new user signup (#1723)

**Comment by ianb** [comment]
I think this is ready for another pass, #1723 and #1364 block a fully successful flow, but don't affect 123done. I think we just need 123done to link to `{CONTENT_SERVER}/subscriptions/products/123doneProProduct?email={email}` (right now it has some text like `Subscribe for pro! LINK TBD! productId=123doneProProduct&`).

**Comment by lmorchard** [comment]
A few quick notes for what's already there in 123done & friends...

* Here is how 123done checks for a subscription capability named `123donePro`: https://github.com/mozilla/fxa/blob/master/packages/123done/static/js/123done.js#L14
* `isSubscribed()` is used to [show or hide](https://github.com/mozilla/fxa/blob/master/packages/123done/static/js/123done.js#L48) an element [#subscriptionStatus](https://github.com/mozilla/fxa/blob/master/packages/123done/static/index.html#L109) element - right now it just says "PRO!" if a user has the capability. This could be made to look fancier - a gold star or something, I don't know.
* `isSubscribed()` is used to [show or hide](https://github.com/mozilla/fxa/blob/master/packages/123done/static/js/123done.js#L65) an element [#subscriptionCTA](https://github.com/mozilla/fxa/blob/master/packages/123done/static/index.html#L119) element - right now it just says "Link TBD" if a user *does not* have the capability. This should link to a content-server redirect URL as implemented for Issue #926

**Comment by lmorchard** [comment]
To be specific about the server-side stuff:

1. We need a Stripe product ID for 123done - e.g. `prod_8675309`
1. `dcdb5ae7add825d2` appears to be the OAuth client ID for 123done - at least, [that's what it is in `dev.json` for oauth-server](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/fxa-oauth-server/config/dev.json#L11).  
1. `123donePro` is the subscription capability name that [123done is looking for to determine "pro" status](https://github.com/mozilla/fxa/blob/master/packages/123done/static/js/123done.js#L8)
1. We need ops to tweak fxa-auth-server config `SUBSCRIPTION_PRODUCT_CAPABILITIES` to include `"prod_8675309": [ "123donePro" ]` - where `prod_8675309` is the Stripe product ID we figure out
1. We need ops to tweak fxa-auth-server config ` SUBSCRIPTIONS_CLIENT_CAPABILITIES` to include `"dcdb5ae7add825d2": [ "123donePro" ]` - where `dcdb5ae7add825d2` is the 123done OAuth client ID

Again, `prod_8675309` is not a real value - we need to get one from Stripe, either from an existing test product or from a newly created one

**Comment by lmorchard** [comment]
And, now that we have a product ID, this should eventually work as the URL to the payment page on the dev server environment:

https://latest.dev.lcip.org/subscriptions/products/prod_Ex9Z1q5yVydhyk

We might want to make the base URL here configurable on 123done. (i.e. `https://latest.dev.lcip.org/subscriptions/products/{product_id}`)

**Comment by clouserw** [comment]
+cc @jrgm @jbuck for the above couple comments about config changes

**Comment by clouserw** [comment]
@dannycoates would you take a look at this?
